internal/provider/resource_aci_pim_route_map_policy.go

```diff
@@ -0,0 +1,547 @@
+			if !tagAnnotation.Key.IsUnknown() {
+				childMap["attributes"]["key"] = tagAnnotation.Key.ValueString()
+			}
+			if !tagAnnotation.Value.IsUnknown() {
+				childMap["attributes"]["value"] = tagAnnotation.Value.ValueString()
+			}
+			childPayloads = append(childPayloads, map[string]interface{}{"tagAnnotation": childMap})
+			tagAnnotationIdentifier := TagAnnotationIdentifier{}
+			tagAnnotationIdentifier.Key = tagAnnotation.Key
+			tagAnnotationIdentifiers = append(tagAnnotationIdentifiers, tagAnnotationIdentifier)
+		}
+		for _, tagAnnotation := range tagAnnotationState {
+			delete := true
+			for _, tagAnnotationIdentifier := range tagAnnotationIdentifiers {
+				if tagAnnotationIdentifier.Key == tagAnnotation.Key {
+					delete = false
+					break
+				}
+			}
+			if delete {
+				childMap := map[string]map[string]interface{}{"attributes": {}}
+				childMap["attributes"]["status"] = "deleted"
+				childMap["attributes"]["key"] = tagAnnotation.Key.ValueString()
+				childPayloads = append(childPayloads, map[string]interface{}{"tagAnnotation": childMap})
+			}
+		}
+	} else {
+		data.TagAnnotation = types.SetNull(data.TagAnnotation.ElementType(ctx))
+	}
+
+	return childPayloads, "", ""
+}
+
+func getPimRouteMapPolCreateJsonPayload(ctx context.Context, data *PimRouteMapPolResourceModel, tagAnnotationPlan, tagAnnotationState []TagAnnotationPimRouteMapPolResourceModel) (*container.Container, string, string) {
+	payloadMap := map[string]interface{}{}
+	payloadMap["attributes"] = map[string]string{}
+	childPayloads := []map[string]interface{}{}
+
+	TagAnnotationchildPayloads, errMessage, errMessageDetail := getPimRouteMapPolTagAnnotationChildPayloads(ctx, data, tagAnnotationPlan, tagAnnotationState)
+	if TagAnnotationchildPayloads == nil {
+		return nil, errMessage, errMessageDetail
+	}
+	childPayloads = append(childPayloads, TagAnnotationchildPayloads...)
+
+	payloadMap["children"] = childPayloads
+	if !data.Annotation.IsNull() && !data.Annotation.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["annotation"] = data.Annotation.ValueString()
+	}
+	if !data.Descr.IsNull() && !data.Descr.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["descr"] = data.Descr.ValueString()
+	}
+	if !data.Name.IsNull() && !data.Name.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["name"] = data.Name.ValueString()
+	}
+	if !data.NameAlias.IsNull() && !data.NameAlias.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["nameAlias"] = data.NameAlias.ValueString()
+	}
+	if !data.OwnerKey.IsNull() && !data.OwnerKey.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["ownerKey"] = data.OwnerKey.ValueString()
+	}
+	if !data.OwnerTag.IsNull() && !data.OwnerTag.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["ownerTag"] = data.OwnerTag.ValueString()
+	}
+
+	payload, err := json.Marshal(map[string]interface{}{"pimRouteMapPol": payloadMap})
```

Comment:
Just a suggestion->
I saw this in a different provider from ours, shall we create a custom Handle Error?

Something like this..

func HandleError(ctx context.Context, err error, message string) diag.Diagnostics {
    var diags diag.Diagnostics

    if err != nil {
        diags = append(diags, diag.Diagnostic{
            Severity: diag.Error,
            Summary:  "Error occurred",
            Detail:   message,
            AttributePath: diag.NewAttributePath().WithAttributeName(Name of the function where error occured),
        })
    }
    return diags
}

Reply:
Actually was something on my list to change, where I would pass the diags to the functions. Have changed now see commit: https://github.com/CiscoDevNet/terraform-provider-aci/pull/1113/commits/ac1dcb81edbd07e752c40cf1221e4bb2c43e012e